kinder/pkg/cri/nodes/containerd/alterhelper.go

```diff
@@ -138,13 +140,142 @@ func ImportImage(bc *bits.BuildContext, tar string) error {
 	return nil
 }
 
-// PreLoadInitImages preload images required by kubeadm-init into the containerd runtime that exists inside a kind(er) node
+// PreLoadInitImages preloads images from .tar files in srcFolder into containerd,
+// renaming images to remove architecture suffixes matching the host architecture.
 func PreLoadInitImages(bc *bits.BuildContext, srcFolder string) error {
-	// NB. this code is an extract from "sigs.k8s.io/kind/pkg/build/node"
-	return bc.RunInContainer(
-		"bash", "-c",
-		`find `+srcFolder+` -name *.tar -print0 | xargs -0 -n 1 -P $(nproc) ctr --namespace=k8s.io images import --all-platforms --no-unpack && rm -rf `+srcFolder+`/*.tar`,
-	)
+	// Get the host architecture.
+	archLines, err := bc.CombinedOutputLinesInContainer("uname", "-m")
+	if err != nil {
+		return fmt.Errorf("failed to get host architecture: %w", err)
+	}
+	if len(archLines) == 0 {
+		return fmt.Errorf("failed to get host architecture: empty output")
+	}
+
+	hostArch := strings.TrimSpace(archLines[0])
+	if hostArch == "" {
+		return fmt.Errorf("failed to get host architecture: empty architecture string")
+	}
+
+	// Map architecture names to standard names
+	archMap := map[string]string{
+		"x86_64":  "amd64",
+		"amd64":   "amd64",
+		"aarch64": "arm64",
+		"arm64":   "arm64",
+	}
+
+	standardArch, exists := archMap[hostArch]
+	if !exists {
+		log.Warnf("Warning: unsupported architecture %s, proceeding with original name\n", hostArch)
+		standardArch = hostArch
+	}
+
+	// List all .tar files in srcFolder inside the container.
+	tarFilesLines, err := bc.CombinedOutputLinesInContainer("find", srcFolder, "-name", "*.tar")
+	if err != nil {
+		return fmt.Errorf("failed to list .tar files: %w", err)
+	}
+
+	// Filter out empty lines and check if we have any files
+	var validTarFiles []string
+	for _, tarFilePath := range tarFilesLines {
+		tarFilePath = strings.TrimSpace(tarFilePath)
+		if tarFilePath != "" {
+			validTarFiles = append(validTarFiles, tarFilePath)
+		}
+	}
+
+	if len(validTarFiles) == 0 {
+		log.Infof("No .tar files found in %s\n", srcFolder)
+		return nil
+	}
+
+	log.Infof("Found %d .tar files to import\n", len(validTarFiles))
+
+	var importErrors []string
+	for _, tarFilePath := range validTarFiles {
+		log.Infof("Importing image from: %s\n", tarFilePath)
+
+		// Import the image
+		importCmd := fmt.Sprintf("ctr --namespace=k8s.io images import --all-platforms '%s'", tarFilePath)
+		importLines, err := bc.CombinedOutputLinesInContainer("bash", "-c", importCmd)
+		if err != nil {
+			errMsg := fmt.Sprintf("failed to import image from %s: %v", tarFilePath, err)
+			log.Errorf("Error: %s\n", errMsg)
+			importErrors = append(importErrors, errMsg)
+			continue // Continue with next file instead of failing completely
+		}
+
+		// Parse the imported image name from the output
```

Comment:
reading the image name from the tar seems less volatile to me.
the log output of ctr has no "API" guarantees and can change at any time.